File: 2025-12-12_Practics/src/practics.py
# 기명 함수 (Named Function)
def greet_Return(name):
    result = "Hello, " + name + "!"
    return result

# print(value)    # 출력: Hello, Alice!
# return 함수: 값을 반환하는 함수

def greet_nonReturn(name): # return문 없음 : 반환값이 any -> none (함수명에 마우스 오버로 확인)
   print("Hello, " + name + "!")
   # return "return-none" # 주석 박으면 자동으로 none 리턴, 터미널에 출력

# 인간의 내장과 같은 함수의 필수 3요소: 입력(매개변수등), 논리/알고리즘이 들어간 기능과 기능의 실행, 출력(산출물)


def routine():
    def 일어나기(time):
        시간 = time
        return 시간
    
    def 옷입기(상의, 하의, 아우터):
        옷 = { "상의" : 상의,
             "하의" : 하의,
             "아우터" : 아우터
             }
        return 옷
        

File: 2025-12-12_Practics/practics.js
// 기명 함수, return 함수

// 기명 함수 (Named Function)
function greet(name) {
    return "Hello, " + name + "!";
}
println(greet("Alice"));    // 출력: Hello, Alice!
// return 함수: 값을 반환하는 함수

// --------------------------------

// 다른 함수들
// 익명 함수 (Anonymous Function) - 함수 표현식
const farewell = function(name) {
    return "Goodbye, " + name + "!";
};
println(farewell("Bob"));   // 출력: Goodbye, Bob!

// 화살표 함수 (Arrow Function) - 익명 함수의 또 다른 형태
const add = (a, b) => a + b;
println(add(5, 3));        // 출력: 8
// 기명 함수는 함수 이름이 명시되어 있어 재귀 호출이나 디버깅에 유리합니다.
// 익명 함수는 주로 일회성 작업이나 콜백 함수로 사용됩니다.
// 화살표 함수는 간결한 문법을 제공하며, this 바인딩이 다르게 동작합니다.

// 실행 함수 (Execution function, exe)
// 실행만 시키고 종료하는 함수 = 서버 가동 후 종료되는 함수